### ISSUE TITLE
Frostmounre server specific - free weapon skill up on level up lost on log out past 200 weapon skill

### ISSUE THREAD

**Issue by devriesp9000** [gamefreedomgit/Frostmourne#2684]
**Description:**
While playing on Frostmourne, upon level up you get all your weapon skills (and skill caps) increased. Upon log out, this is lost past 200 weapon skill

**How to reproduce:**
1. Create a new account on Whitemane
2. Download and install wow 3.3.5
3. Login and create a character on Frostmourne
4. Load in to the game world
5. Level your character to level 41 or higher
6. Check your weapon skills upon level up (they are correct)
![2024_12_31_15_00_FreeSkillUp](https://github.com/user-attachments/assets/37f82bc7-3522-4bb7-82e7-4ac5a46a1621)
(screenshot taken on December 31, 2024 - 3pm CET)
7. Log out
8. Login and load back in to the game world
9. check your weapon skills - they have been lowered to 200 (weapon skill cap is correct)
![2024_12_31_15_00_SkillUpLost](https://github.com/user-attachments/assets/49066c38-f169-49cf-9d25-da651e379e86)
(screenshot taken on December 31, 2024 - 3:02pm CET

**How it should work:**
1. Create a new account on Whitemane
2. Download and install wow 3.3.5
3. Login and create a character on Frostmourne
4. Load in to the game world
5. Level your character to level 41 or higher
6. Check your weapon skills upon level up (they are correct)
7. Log out
8. Login and load back in to the game world
9. check your weapon skills remain at the level they were set to

**Database links:**
n/a

**Comment by Riztazz** [comment]
What was your characters name?

**Comment by Iconostasis** [comment]
Badjanaka - 76 lvl blood elf hunter - same experience with the auto skill-up feature.
Any time I level up it boots everything up properly and when I relog it goes back to what I previously had.

**Comment by Riztazz** [comment]
Same question to you:P do you use equipment sets by any chance, and some of them are longer than 16 characters?

**Comment by Iconostasis** [comment]
No, I do not use the built-in feature. I also don't have addons for that.

**Comment by devriesp9000** [comment]
> What was your characters name? Any by any chance did you create an equipment set for your character with long name?

Character name is Huntzor (BE hunter) and I am not using any equipment sets

**Comment by devriesp9000** [comment]
Hello, I re-tested today and it does still occur, that said, the skills are now "stuck" at 280 (not 200 as before) so something did occur.
![2025_1_6_16_07SkillUpLost](https://github.com/user-attachments/assets/bd4b6d72-0d18-410a-a457-5821260ab98b)

**Comment by Riztazz** [comment]
Tagging https://github.com/gamefreedomgit/Frostmourne/issues/1818 related

**Comment by Riztazz** [comment]
Tagging https://github.com/gamefreedomgit/Frostmourne/issues/1679 related

**Comment by Riztazz** [comment]
Can you check if it happens now? And if so, please tell me the exact time you did check so i can cross-check logouts

**Comment by devriesp9000** [comment]
> Can you check if it happens now? And if so, please tell me the exact time you did check so i can cross-check logouts

I retested, I leveled up at 9:06am CET and then logged out and back in at 9:10 CET - the problem persists for me

**Comment by devriesp9000** [comment]
sorry, closed the issue by accident -_-'

**Comment by Riztazz** [comment]
All good, i can't track what's happening to your character, i'm just gonna make an exact copy and see locally. Fixed a bunch of issues but yours still persist:P

**Comment by devriesp9000** [comment]
weird, I don't know if it matters, but i'm using the HD client

**Comment by Riztazz** [comment]
I'll figure it out, sorry it's taking a bit long :) Thanks for the info!

**Comment by devriesp9000** [comment]
@Riztazz - I have another character where I can reproduce the issue - level 56 Troll DK. Gained a level, logged out, logged back in and weapon skills are back to 270 (out of 280)

**Comment by Riztazz** [comment]
What is the name?

**Comment by devriesp9000** [comment]
Char name is Deathzor

**Comment by Riztazz** [comment]
I am happy to report that i've finally managed to reproduce this stupid issue, i'll fix it in a bit. Will close once it's live:P

**Comment by devriesp9000** [comment]
@Riztazz - i have very bad news, the issue is back :( it was OK for a few levels (think 1 week) for my hunter (Huntzor BE level 70) but now i'm having the exact same issue again

**Comment by Riztazz** [comment]
are fucking kidding me, wtf

**Comment by devriesp9000** [comment]
@Riztazz ok, here is where it gets weird. On level 69 -70 I logged out (there was a server update) and the i logged in a few hours later and the problem was back. Just now I leveled up 70 - 71 and it's not a problem again :/

**Comment by devriesp9000** [comment]
Latest updated - I just leveled up again (71 - 72) and the problem is reproducible again. just did a quick logout -> login for testing

**Comment by Riztazz** [comment]
Which character is it?

**Comment by devriesp9000** [comment]
> Which character is it?

@Riztazz still on my BE hunter "Huntzor"

**Comment by Riztazz** [comment]
I am close to figuring out what's up with your character again :P

**Comment by Riztazz** [comment]
Eh i can't find a reproduction, your character always saves for me.
Do you use racial swap feature or something like that? Can you also try after most recent update?
